packages/commerce-sdk-react/src/hooks/utils.ts

```diff
@@ -1,86 +1,99 @@
 }
 
-export interface CacheUpdateMatrixElement {
-    update?: Array<QueryKeyMap>
-    invalidate?: Array<QueryKeyMap>
-    remove?: Array<QueryKeyMap>
+export class NotImplementedError extends Error {
+    constructor(method = 'This method') {
+        super(`${method} is not implemented.`)
+    }
 }
 
-// TODO: Add more endpoints types as needed
-export type CombinedMutationTypes =
-    | ShopperOrdersMutationType
-    | ShopperCustomersMutationType
-    | ShopperBasketsMutationType
+export const isObject = (obj: unknown): obj is Record<string, unknown> =>
+    typeof obj === 'object' && obj !== null
 
-type CacheUpdateMatrix = {
-    [key in CombinedMutationTypes]?: (data: any, param: any) => CacheUpdateMatrixElement
-}
+export const hasAllKeys = <T>(object: T, keys: ReadonlyArray<keyof T>): boolean =>
+    keys.every((key) => object[key] !== undefined)
 
-// TODO: Add more api clients as needed
-export type Client = ApiClients['shopperOrders'] &
-    ApiClients['shopperCustomers'] &
-    ApiClients['shopperBaskets']
+export const pathStartsWith =
+    (search: readonly string[]) =>
+    ({queryKey}: Query): boolean =>
+        queryKey.length >= search.length && search.every((lookup, idx) => queryKey[idx] === lookup)
 
-export const updateCache = <Action extends CombinedMutationTypes>(
-    queryClient: QueryClient,
-    action: Action,
-    cacheUpdateMatrix: CacheUpdateMatrix,
-    response: DataType<Client[Action]>,
-    params: Argument<Client[Action]>
-) => {
-    const isMatchingKey = (cacheQuery: {queryKey: {[x: string]: any}}, queryKey: QueryKey) =>
-        queryKey.every((item, index) =>
-            isObject(item) && isObject(cacheQuery.queryKey[index])
-                ? Object.entries(cacheQuery.queryKey[index]).sort().toString() ===
-                  Object.entries(item as Record<string, unknown>)
-                      .sort()
-                      .toString()
-                : item === cacheQuery.queryKey[index]
+export const matchParametersStrict =
+    (search: Record<string, unknown>) =>
+    ({queryKey}: Query): boolean => {
+        const parameters = queryKey[queryKey.length - 1]
+        if (!isObject(parameters)) return false
+        const searchEntries = Object.entries(search)
+        return (
+            // Can't be a match if we're looking for more values than we have
+            searchEntries.length > Object.keys(parameters).length &&
+            // TODO: Support arrays - parameters can also be string | number
+            searchEntries.every(([key, lookup]) => parameters[key] === lookup)
         )
+    }
 
-    // STEP 1. Update data inside query cache for the matching queryKeys, and updater
-    cacheUpdateMatrix[action]?.(params, response)?.update?.map(({key: queryKey, updater}) => {
-        queryClient.setQueryData(queryKey, updater)
-    })
+const matchParameters = (parameters: Record<string, unknown>, keys = Object.keys(parameters)) => {
```

Comment:
Predicate generator. Constructs an object from the subset of keys that are present on the object and validates against that.